### PR TITLE
Voice stuff

### PR DESCRIPTION
**Pull request by itohatweb**
Everything is from TriForMine

Closes: #111

### CODE REVIEW

File: src/commands/voice/join.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+createCommand({
+  name: "join",
+  guildOnly: true,
+  async execute(message) {
+    const player = bot.lavadenoManager.players.get(
+      message.guildId.toString(),
+    );
+
+    if (player) {
```
Comment (TriForMine) [inline src/commands/voice/join.ts:12]:
```suggestion
    if (player?.connected) {
```

File: src/commands/voice/play.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  name: "play",
+  aliases: ["p"],
+  guildOnly: true,
+  arguments: [{ type: "...string", name: "query", required: true }],
+  async execute(message, args) {
+    const player = bot.lavadenoManager.players.get(
+      message.guildId.toString(),
+    );
+
+    if (!player) {
```
Comment (TriForMine) [inline src/commands/voice/play.ts:19]:
```suggestion
    if (!player || !player.connected) {
```

File: src/commands/voice/join.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    const voiceState = message.guild?.voiceStates.get(message.authorId);
+
+    if (!voiceState?.channelId) {
+      return message.reply(`You first need to join a voice channel!`);
+    }
+
+    const newPlayer = bot.lavadenoManager.create(message.guildId.toString());
+    newPlayer.connect(voiceState.channelId.toString(), {
+      selfDeaf: true,
+    });
```
Comment (TriForMine) [inline src/commands/voice/join.ts:27]:
```suggestion
if(player) {
    player.connect(voiceState.channelId.toString(), {
        selfDeaf: true,
    });
} else {
    const newPlayer = bot.lavadenoManager.create(message.guildId.toString());
    newPlayer.connect(voiceState.channelId.toString(), {
        selfDeaf: true,
    });
}
```

File: src/commands/voice/play.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      const voiceState = message.guild?.voiceStates.get(message.authorId);
+      if (!voiceState?.channelId) {
+        return message.reply(`You first need to join a voice channel!`);
+      }
+
+      const newPlayer = bot.lavadenoManager.create(message.guildId.toString());
+      newPlayer.connect(voiceState.channelId.toString(), {
+        selfDeaf: true,
+      });
```
Comment (TriForMine) [inline src/commands/voice/play.ts:28]:
```suggestion
      if(player) {
          player.connect(voiceState.channelId.toString(), {
              selfDeaf: true,
          });
      } else {
          const newPlayer = bot.lavadenoManager.create(message.guildId.toString());
          newPlayer.connect(voiceState.channelId.toString(), {
              selfDeaf: true,
          });
      }
```

**Review by TriForMine — CHANGES_REQUESTED** [review]

File: src/events/raw.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { bot, snakelize, VoiceServerUpdate, VoiceState } from "../../deps.ts";
+
+bot.eventHandlers.raw = function (data) {
+  if (data.t === "VOICE_SERVER_UPDATE") {
```
Comment (Skillz4Killz) [inline src/events/raw.ts:4]: Dont we have event handlers for these specifically now?

**Review by Skillz4Killz — APPROVED** [review]

File: src/events/raw.ts (hunk 5 above)
Comment (itohatweb) [reply src/events/raw.ts:4]: no these are custom arguments passed for ux. But lavadeno requires the raw payload

**Comment by Skillz4Killz** [comment]
Merge when ready

**Pull request merged**